File: cpp/Blind75/ArrayHash/TwoSum.cpp
class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            int size = nums.size();
            vector<int> pivos(2);
            for (int i = 0; i < size; ++i) {
                for (int j = i + 1; j < size; ++j) {
                    if (nums[i] + nums[j] == target) {
                        pivos[0] = i;
                        pivos[1] = j;
                        return pivos;
                    }
                }
            }
        }

        vector<int> twoSumOpt(vector<int>& nums, int target) {
            std::map<int, int> m;
    
            for (int i = 0; i < nums.size(); ++i) {
                int com = target - nums[i];
                if (m.find(com) != m.end()) {
                    return {m[com], i};
                }
    
                m[nums[i]] = i;
            }
        }
    };
    

File: Python/Grind75/TwoSum.py
from typing import List

# https://leetcode.com/problems/two-sum/


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)

        for i in range(n):
            for j in range(i + 1, n):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []


class SolutionHashmap:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}

        for idx, num in enumerate(nums):
            remain = target - num
            if remain in m:
                return [idx, m[remain]]
            m[num] = idx
        return []


if __name__ == "__main__":
    solution = Solution()
    print(solution.twoSum([2, 7, 11, 15], 9))
